### PR TITLE
Initial poll support

### PR DESCRIPTION
**Pull request by applenick**
# Polls

(Will writeup details soon)

Signed-off-by: applenick <applenick@users.noreply.github.com>

### CODE REVIEW

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -97,6 +95,13 @@ freeze:
 mutations:
   enabled: true  
   
+# Polls - Allow players to decide the outcome of question via an interactive process
+polls:
+  enabled: true 
+  default-time: 60  # Default length (in seconds) of a poll
+  confirm: true     # Whether to confirm final outcome with staff member
```
Comment (applenick) [inline src/main/resources/config.yml:102]: Note to self: oops, forgot to implement confirmation command.

**Comment by applenick** [comment]
Feedback:
- Allow poll creation via single command
- Show votes in bossbar
- reduce broadcasts (at beginning and end only)

File: src/main/java/dev/pgm/community/poll/Poll.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+   */
+  int getVoteTally(boolean yes);
+
+  /**
+   * Get the player executing the poll
+   *
+   * @return A player if online (or null)
+   */
+  @Nullable
+  Player getExecutor();
```
Comment (Pablete1234) [inline src/main/java/dev/pgm/community/poll/Poll.java:32]: Should be CommandSender, console should be able to poll too

File: src/main/java/dev/pgm/community/poll/PollBase.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  @Override
+  public boolean getOutcome() {
+    return getVoteTally(true) > getVoteTally(false);
+  }
+
+  @Override
+  public int getVoteTally(boolean yes) {
+    return Math.toIntExact(votes.values().stream().filter(vote -> yes ? vote : !vote).count());
```
Comment (Pablete1234) [inline src/main/java/dev/pgm/community/poll/PollBase.java:98]:
```suggestion
    return Math.toIntExact(votes.values().stream().filter(vote -> vote == yes).count());
```

**Comment by applenick** [comment]
Had an idea, will look into displaying a cool GUI

**Comment by applenick** [comment]
Outdated, will be revitalized in the upcoming `party mode` feature set.

**Pull request closed without merging**